hax/hax/util.py

```diff
@@ -446,9 +446,54 @@ def get_service_keys(node_items: List[Any], fidk: int) -> List[Any]:
             if f'{fid}' == x['Key'].split('/')[-1]
         ]
 
+    def is_node_alive(self, node: str) -> bool:
+        """
+        Checks via Consul Members API whether the given node is alive.
+        """
+        try:
+            # Returns data of the following kind:
+            # [{
+            #     'Name': 'localhost',
+            #     'Addr': '192.168.6.214',
+            #     'Port': 8301,
+            #     'Tags': {
+            #         'acls': '0',
+            #         'bootstrap': '1',
+            #         'build': '1.7.8:9a5a1218',
+            #         'dc': 'dc1',
+            #         'id': 'dd8a91f6-ca32-30e0-983c-8f309d653045',
+            #         'port': '8300',
+            #         'raft_vsn': '3',
+            #         'role': 'consul',
+            #         'segment': '',
+            #         'vsn': '2',
+            #         'vsn_max': '3',
+            #         'vsn_min': '2',
+            #         'wan_join_port': '8302'
+            #     },
+            #     'Status': 1,
+            #     'ProtocolMin': 1,
+            #     'ProtocolMax': 5,
+            #     'ProtocolCur': 2,
+            #     'DelegateMin': 2,
+            #     'DelegateMax': 5,
+            #     'DelegateCur': 4
+            # }]
+            members_data = self.cns.agent.members()
+            LOG.log(TRACE, "members: %s", members_data)
+            for member in members_data:
+                if member['Name'] == node:
+                    return int(member['Status']) == 1
+            return False
+        except (ConsulException, HTTPError, RequestException) as e:
+            raise HAConsistencyException(
+                'Failed to members data from Consul') from e
+
     def get_node_health(self, node: str) -> str:
         try:
             node_data = self.cns.health.node(node)[1]
+            if not node_data and (not self.is_node_alive(node)):
+                return 'failed'
             return str(node_data[0]['Status'])
```

Comment:
What if the node is alive, but the health for it is `[]` (for some reason) - how will it work?

Reply:
It will be 'failed'

Reply:
But if the node is alive the check for `not self.is_node_alive(node)` will be false and the whole `if` will be false - how come it will be `failed`?

Reply:
Ah, now I get your question. The situation when `node_data == []` AND `self.is_node_alive(node) == True` would cause IndexError exception.

This would mean that Consul Health API doesn't know our node but at the same time Consul membership information contains this node. So there should be a kind of inconsistency between two different Consul APIs. I think it is fine to assume that two Consul APIs are consistent.

The only problem is that there is a potential time gap between `node_data = self.cns.health.node(node)[1]` call and `             members_data = self.cns.agent.members()` (happens inside is_node_alive()). Because of the time factor, the truth can really change and this situation is potentially possible.

Will it be sane to replace this if statement like this?
```python
            if not node_data or (not self.is_node_alive(node)):
                return 'failed'
```

@mssawant what do you think?

Reply:
I've changed the if statement (and -> or).
The PR is rebased against the latest main.

Reply:
@andriytk Can you please review?

Reply:
Looks fine to me now.